### PR TITLE
Handle security error in Firefox when cookie is disabled

### PR DESCRIPTION
**Pull request by chrmod**
fixes: https://github.com/dfahlander/Dexie.js/issues/617 and https://github.com/dfahlander/Dexie.js/issues/614

we've submited related fixes to PouchDB which were already merged: https://github.com/pouchdb/pouchdb/pull/6890

### CODE REVIEW

**Comment by dfahlander** [comment]
Thanks! I will rewrite the PR a bit just for optimization.

File: src/Dexie.js [hunk 1]
```diff
@@ -3152,8 +3152,17 @@ props(Dexie, {
...
+            try {
+                return _global.indexedDB || _global.mozIndexedDB || _global.webkitIndexedDB || _global.msIndexedDB;    
+            } catch(e) {}
+            
+        },
+        get IDBKeyRange() {
+            try {
+                return _global.IDBKeyRange || _global.webkitIDBKeyRange;                
+            } catch(e) {}
+        }
```
Comment (gregor) [inline src/Dexie.js:3165]: @dfahlander Thanks for fixing #617 a few days back (which is the bottom part of this PR). This code change above is still needed for the same issue in Firefox though. Can I ask you again for a fix and release on the 2.x branch please? Disabled cookies still throw a security error on load of dexie thus resulting in a blank page on https://app.wire.com. (https://github.com/wireapp/wire-webapp)

**Comment by dfahlander** [comment]
Sorry for taking some time. The changes are now in both master-2 and master. New 2.0.4 release coming soon.

**Comment by gregor** [comment]
Thank you very much for taking care of this. Happy to see the update coming back from vacation. 🎉

**Pull request closed without merging**